docs/DeveloperGuide.adoc

```diff
@@ -153,94 +153,234 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+====
+
+The model will then be updated using the predicate obtained from the `getAreaPredicate()` function.
+
+A new `CommandResult` will be created and returned to the `LogicManager`.
+
+[[import]]
+=== Import feature
+The import feature allows users to save the trouble of adding the delivery orders and the return orders one by one
+when they have large amount of delivery orders or return orders to add into Delino.
 
-The following sequence diagram shows how the undo operation works:
+The import feature was implemented as a `ImportCommand` in the `Logic` package.
 
-image::UndoSequenceDiagram.png[]
+==== Class Diagram for Import feature:
 
-NOTE: The lifeline for `UndoCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+.Import Command Class Diagram
+image::ImportClassDiagram.png[]
 
-The `redo` command does the opposite -- it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the address book to that state.
+==== Sequence Diagram for importing the csv file.
+The sequence diagram for the *Import Command* when import the data from the  *csv file*
+is shown below:
+
+.Import Command Sequence Diagram
+image::ImportCommandSequenceDiagram.png[]
+
+The arguments passed to the Import Command will be parsed by the `ImportCommandParser` class. +
+If the given arguments are valid, a new `ImportCommand` object will be returned.
+In this class, invalid arguments will result in a `ParseException`. +
+
+Two types of checks will be done before returning the `ImportCommand`:
+
+* Check whether the file is exists in the data folder.
+* Check whether the first prefix is `orderType` prefix.
 
 [NOTE]
-If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone address book states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+====
+`CsvProcessor` is  a helper class that helps to retrieve the data from the csv file and process the data before
+giving to `ImportCommand`.
+====
+
+After the `ImportCommand` object is being returned to the `LogicManager`, the `LogicManager` will start
+to run the execute the import Command, this is will be shown at the <<Execution of Import Command Sequence Diagram, Execution of Import Command Sequence Diagram>>.
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
+==== Execution of Import Command Sequence Diagram
 
-image::UndoRedoState4.png[]
+.Import Command Sequence Diagram 2
+image::ImportCommandSequenceDiagram2.svg[]
```

Comment:
Would the rightmost anonymous object be required? It's good that you have referenced this diagram from the previous page, but the previous diagram doesn't seem to have that object